src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -500,6 +597,147 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+                    var interpolationData = interpolatedString.InterpolationData.GetValueOrDefault();
+
+                    BoundLocal? trailingOutVariable = null;
+                    var constructor = interpolationData.GetValidConstructor();
+                    var lastArgument = constructor.Arguments[^1];
+                    if (interpolationData.HasTrailingHandlerValidityParameter)
+                    {
+                        var outValidityTemp = _factory.SynthesizedLocal(lastArgument.Type!, lastArgument.Syntax);
+                        temporariesBuilder.Add(outValidityTemp);
+                        trailingOutVariable = _factory.Local(outValidityTemp);
+                    }
+
+                    foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                    {
+                        // We can have duplicates because arguments can be duplicated, but we
+                        // only need to register one placeholder per argument
+                        if (HasPlaceholderReplacement(placeholder))
+                        {
+                            continue;
+                        }
+
+                        switch (placeholder.ArgumentIndex)
+                        {
+                            case BoundInterpolatedStringArgumentPlaceholder.InstanceParameter:
+                                Debug.Assert(requiresInstanceReceiver);
+                                Debug.Assert(visitedReceiver != null);
+                                AddPlaceholderReplacement(placeholder, visitedReceiver);
+                                break;
+                            case BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter:
+                                Debug.Assert(trailingOutVariable != null);
+                                AddPlaceholderReplacement(placeholder, trailingOutVariable);
+                                break;
+                            case >= 0 and var argumentIndex:
+                                Debug.Assert(argumentIndex < interpolatedStringHandlerArgumentIndex);
+                                var placeholderParamIndex = argsToParamsOpt.IsDefault ? argumentIndex : argsToParamsOpt[argumentIndex];
+                                Debug.Assert(actualArguments[placeholderParamIndex] is not null);
+                                AddPlaceholderReplacement(placeholder, actualArguments[placeholderParamIndex]);
+                                break;
+                            default:
+                                throw ExceptionUtilities.UnexpectedValue(placeholder.ArgumentIndex);
+                        }
+                    }
+
+                    var (handlerExpressions, result) = RewriteToInterpolatedStringHandlerPattern(interpolatedString);
+
+                    foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                    {
+                        if (HasPlaceholderReplacement(placeholder))
```

Comment:
Consider either removing the assert in `RemovePlaceholderReplacement` or add another method like `RemovePlaceholderReplacementIfPresent(...)`